sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -8873,6 +8873,88 @@ pi_result piextUSMGetMemAllocInfo(pi_context Context, const void *Ptr,
   return PI_SUCCESS;
 }
 
+/// API for writing data from host to a device global variable.
+///
+/// \param Queue is the queue
+/// \param Program is the program containing the device global variable
+/// \param Name is the unique identifier for the device global variable
+/// \param BlockingWrite is true if the write should block
+/// \param Count is the number of bytes to copy
+/// \param Offset is the byte offset into the device global variable to start
+/// copying
+/// \param Src is a pointer to where the data must be copied from
+/// \param NumEventsInWaitList is a number of events in the wait list
+/// \param EventWaitList is the wait list
+/// \param Event is the resulting event
+pi_result piextEnqueueDeviceGlobalVariableWrite(
+    pi_queue Queue, pi_program Program, const char *Name, pi_bool BlockingWrite,
+    size_t Count, size_t Offset, const void *Src, pi_uint32 NumEventsInWaitList,
+    const pi_event *EventsWaitList, pi_event *Event) {
+  PI_ASSERT(Queue, PI_ERROR_INVALID_QUEUE);
+
+  std::scoped_lock<pi_shared_mutex> lock(Queue->Mutex);
+
+  // Find global variable pointer
+  size_t GlobalVarSize = 0;
+  void *GlobalVarPtr = nullptr;
+  ZE_CALL(zeModuleGetGlobalPointer,
+          (Program->ZeModule, Name, &GlobalVarSize, &GlobalVarPtr));
+  PI_ASSERT(GlobalVarSize >= Offset + Count, PI_ERROR_INVALID_VALUE);
```

Comment:
I agree, `PI_ERROR_INVALID_VALUE` isn't very telling here, but it follows the behavior of the interfaces in [cl_intel_global_variable_access](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/opencl-extensions/cl_intel_global_variable_access.asciidoc). I am not opposed to adding a new error for this, but from a API behavior consistency point-of-view it would be easier to document this behavior or change the other plugins (including the OpenCL one) to return the new error too for this failure case. What is your preference?

Reply:
To make it clear, I do not propose adding an ew PI error code, but use existing `PI_ERROR_PLUGIN_SPECIFIC_ERROR`, see here: https://github.com/intel/llvm/blob/ac58dd33126c5e96860dad3b414e9f3ffdf8bd11/sycl/plugins/level_zero/pi_level_zero.cpp#L3098